types/mempool/sender_nonce.go

```diff
@@ -82,7 +107,11 @@ func (snm *senderNonceMempool) Insert(_ sdk.Context, tx sdk.Tx) error {
 		snm.senders[sender] = senderTxs
 	}
 	senderTxs.Set(nonce, tx)
-
+	key := txKey{nonce: nonce, address: sender}
+	_, found = snm.existingTx[key]
+	if !found {
+		snm.existingTx[key] = true
+	}
```

Comment:
At this point, given that it is a map, we can now just simply insert with `true` like this
```go
key := txKey{nonce: nonce, address: sender}
snm.existingTx[key] = true
```